### ISSUE TITLE
Pseudo-Mercator for PCIDSK/PCIDSK dataset

### ISSUE THREAD

**Issue by drnextgis** [OSGeo/gdal#7647]
I'm trying to assign `EPSG:3857` SRS to the PCIDSK/PCIDSK Database File. Apparently the driver doesn't support that SRS and instead of assigning the requested `EPSG:3857` gdal_edit assigns `EPSG:3395` without raising any warnings:

```
$ gdal_edit.py -a_srs "EPSG:3857" target.pix
$ gdalsrsinfo -o epsg target.pix
Confidence in this match: 90 %

EPSG:3395

$ gdalinfo --version            
GDAL 3.6.4, released 2023/04/17

**Comment by warmerdam** [comment]
More specifically, the following SRS is assigned which notably has no *explicit* EPSG relationship recorded for the coordinate system as a whole.  

```
PROJCRS["unnamed",
    BASEGEOGCRS["WGS 84",
        DATUM["World Geodetic System 1984",
            ELLIPSOID["WGS 84",6378137,298.257223563,
                LENGTHUNIT["metre",1]]],
        PRIMEM["Greenwich",0,
            ANGLEUNIT["degree",0.0174532925199433]],
        ID["EPSG",4326]],
    CONVERSION["Mercator (variant A)",
        METHOD["Mercator (variant A)",
            ID["EPSG",9804]],
        PARAMETER["Latitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8801]],
        PARAMETER["Longitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8802]],
        PARAMETER["Scale factor at natural origin",1,
            SCALEUNIT["unity",1],
            ID["EPSG",8805]],
        PARAMETER["False easting",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8806]],
        PARAMETER["False northing",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8807]]],
    CS[Cartesian,2],
        AXIS["(E)",east,
            ORDER[1],
            LENGTHUNIT["metre",1,
                ID["EPSG",9001]]],
        AXIS["(N)",north,
            ORDER[2],
            LENGTHUNIT["metre",1,
                ID["EPSG",9001]]]]
```
I'm testing with irvine.pix from the gdal data repository as a test file.  If I do:

`gdal_translate irvine.pix irvine.tif -b 1 -a_srs EPSG:3857`

I get a report on the tiff (presumably properly representing EPSG:3857 including:

`Upper Left  (  428720.000, 3734400.000) (  3d51' 4.53"E, 31d46'46.17"N)
`
while for the corresponding pcidsk file I get:

`Upper Left  (  428720.000, 3734400.000) (  3d51' 4.53"E, 31d57' 7.16"N)
`

You can see the latitude results of the coordinate system in the PCIDSK file are quite different than expected.  So I concur that the pseudo-mercator is not being represented properly in the PCIDSK file.

**Comment by warmerdam** [comment]
OK, I've determined EPSG:3857 is represented this way in "wkt1" format (which is used by ogr_srs_pci.cpp):
```

PROJCS["WGS 84 / Pseudo-Mercator",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Mercator_1SP"],
    PARAMETER["central_meridian",0],
    PARAMETER["scale_factor",1],
    PARAMETER["false_easting",0],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["Easting",EAST],
    AXIS["Northing",NORTH],
    EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs"],
    AUTHORITY["EPSG","3857"]]
```
notably the projection method is just Mercator_1SP which ogr_srs_pci.cpp translates to the PCIDSK "MER" (mercator) projection.   

I'm only vaguely recalling how this works, but pseudo-mercator was traditionally handled in OGRSpatialReference with a mercator projection and WGS84 geographic coordinate system, but an override proj string that forced use of a sphere and null nadgrids.

eg.

```
$ gdalsrsinfo  EPSG:3857 -o wkt1
PROJCS["WGS 84 / Pseudo-Mercator",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Mercator_1SP"],
    PARAMETER["central_meridian",0],
    PARAMETER["scale_factor",1],
    PARAMETER["false_easting",0],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["Easting",EAST],
    AXIS["Northing",NORTH],
    EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs"],
    AUTHORITY["EPSG","3857"]]
```

But there is on checking for the special case of pseudo-mercator in the ogr_srs_pci.cpp code.  This could be added, but I'm not actually sure how one should represent pseudo-mercator in PCIDSK format -- to learn that we might need a consult from Catalyst (PCI).

**Comment by warmerdam** [comment]
I have sent a request to Catalyst for a small google mercator sample file to work from.

**Comment by warmerdam** [comment]
the appropriate ellips.txt definition for E020 is:

"E020","WGS 84 semimajor axis",6378137.,6378137.

**Comment by drnextgis** [comment]
Thanks for looking into it @warmerdam! Are you planning to make a PR with the fix?

**Comment by warmerdam** [comment]
I was intending to prepare a PR - likely no sooner than the weekend.  I would also be receptive to a PR from others.

**Comment by rouault** [comment]
PR in https://github.com/OSGeo/gdal/pull/7663

**Comment by warmerdam** [comment]
I confirm that PR appears to have resolved the issue, thanks!

**Comment by drnextgis** [comment]
Thanks for resolving the issue promptly!

**Comment by mccalla-catalyst** [comment]
I did some testing of this issue using **GDAL 3.8.0dev-22ba76df32, released 2023/05/22** and **QGIS 3.31.0 Master**. I have a couple of PIX files that, in PCI-parlance, have mapunit projection strings of **WebMerc D894**. In the older version of QGIS (e.g. QGIS 3.26) my data would be shifted north from other reference data. And now due to this fix, my data does align well in QGIS. This is obviously good news.

But the details of the coordinate system for my PIX files is a little confusing. Now GDALINFO is outputting this:
> PROJCRS["unnamed",
>     BASEGEOGCRS["Unknown - PCI D894",
>      DATUM["Unknown - PCI D894",
>        ELLIPSOID["Unknown - PCI D894", 
>    ...

In my opinion, it would be nice if this CRS had a proper name. [@warmerdam](https://github.com/warmerdam) is this what you expected? I’m not familiar enough with GDAL to have an informed suggestion of what an appropriate name would be. But since the original scenario was defining the coordinate system to **EPSG:3857**, it would, ideally, reflect that. Could GDAL not report the same sort of information from **gdalsrsinfo EPSG:3857 -o wkt1** i.e.:
> PROJCRS["WGS 84 / Pseudo-Mercator",
>    BASEGEOGCRS[“WGS 84",
>      DATUM["WGS_1984",
>        SPHEROID["WGS 84",6378137,298.257223563,**
> ...

**Comment by warmerdam** [comment]
It would be desirable if the PCI datum and ellipsoid definitions included EPSG equivelents and we could just substitute those in.  However that is somewhat beyond the scope of just making the world mercator coordinate system work adequately.   So I think this PR accomplished it's goal, even though we could imagine further improvements to the PCI coordinate system mappings in the future.
